tests/include/alt-dummy/arc4_alt.h

```diff
@@ -0,0 +1,30 @@
+/* arc4_alt.h with dummy types for MBEDTLS_ARC4_ALT */
```

Comment:
When the review here has settled, if #4588 isn't merged yet, I'll rewrite the history to skip modules that are about to be removed in “Create xxx_alt.h headers for testing”.

Reply:
I don't think there will be any merge conflict since this PR doesn't touch places where modules having ALT are referenced. Nor CI failures since this PR doesn't rely on the existence of ALT for modules other than DHM, ECP and threading. Just some useless files which we can remove in a follow-up.

Reply:
Agreed, so I'm marking this conversation as resolved.